src/py42/_internal/archive_access.py

```diff
@@ -159,15 +164,21 @@ def is_job_complete(self, job_id):
         response = self._storage_archive_client.get_restore_status(job_id)
         return self._get_completion_status(response)
 
-    def _start_restore(self, file_selection):
+    def _start_restore(self, file_selection, exceptions):
+        num_files = sum([fs.num_files for fs in file_selection])
+        num_dirs = sum([fs.num_dirs for fs in file_selection])
+        size = sum([fs.size for fs in file_selection])
+        zip_result = _check_for_multiple_files(file_selection) or None
         return self._storage_archive_client.start_restore(
-            self._device_guid,
-            self._archive_session_id,
-            file_selection.path_set,
-            file_selection.num_files,
-            file_selection.num_dirs,
-            file_selection.size,
+            guid=self._device_guid,
+            web_restore_session_id=self._archive_session_id,
+            path_set=[fs.path_set for fs in file_selection],
+            num_files=num_files,
```

Comment:
is `num_files` supposed to include the files that are in the selected directories?

Reply:
`supposed to` ? No, I don't think it really matters.

But does it in the UI? Yes.

In any case, we are setting them to `1` always when crafting `FileSelection` objects. This is because these params don't really matter.  However, when you get to _this_ point in a file restore execution, I'd expect this is where you would amass them the counts for the request.  During the creaetion of `FileSelection` objects is where you would figure out all the counts for `num_files`, `num_dirs`, and `size`.... not here.

Reply:
if they don't matter, why do anything different with them than what we were doing previously?

Reply:
Previously there was only one value each.. Now, there is a list. Do I just use the values of the first element of the list? That seems odd... So I amassed them.

Reply:
I am considering that it might be best to just implement the logic for figuring out the dir and file counts (during the part where we create `FileSelection` tuples). Reasons:

1.  They are required parameters even though they shouldn't be, but why not lie.
2. It could be helpful to know these counts during debugging and looking at sent request params.  (maybe?)
3. It could afford progress bars in people's scripts.

Reply:
It would be nice to have a method available to generate the file count, especially for when we add push restore functionality, as that's where the real benefit is. The desktop client then can show an accurate progress bar and estimated time remaining, etc. 

However, knowing that in some extremely large-scale archives, calculating file counts can take an extremely long time (if a folder has a million files in it, for example). I'd suggest we add a `calculation_timeout` param that would skip waiting for results past a certain time and just start the restore job.

Reply:
I had issues trying to implement a timeout decorator... I could get it to terminate the whole process but couldn't get it to return defaults values without terminating the whole process.

Would it be easier to just have a parameter for ignoring calculating file sizes and then to recommend large restores to use this parameter?

^ Did this on the last commit

Reply:
Wouldn't you use the https://www.crashplan.com/apidocviewer/#WebRestoreFileSizePolling endpoint to poll for file counts/size (like console UI does) and you could just end the polling loop if it goes past the timeout window?

Reply:
Oh and just check the time in the loop? Ya I guess so. I was trying to using signals and what-not to interrupt the thread after a timeout

Reply:
I was trying to just use "/api/WebRestoreFileSize".

Reply:
It is going to take a while to get this in, but I am working on it